data-prepper-core/src/test/java/org/opensearch/dataprepper/pipeline/router/RouterTest.java

```diff
@@ -143,6 +147,40 @@ void route_with_multiple_DataFlowComponents() {
         }
     }
 
+    @Nested
+    class WithUnroutedRecords {
```

Comment:
It would also be good to have a test case where the records are not `Event` objects. In such a case, we should not call the `Consumer` with those records.